darwin/Platform.c

```diff
@@ -464,10 +469,58 @@ bool Platform_getDiskIO(DiskIOData* data) {
    return true;
 }
 
+/* Caution: Given that interfaces are dynamic, and it is not possible to get statistics on interfaces that no longer exist,
+   if some interface disappears between the time of two samples, the values of the second sample may be lower than those of
+   the first one. */
 bool Platform_getNetworkIO(NetworkIOData* data) {
-   // TODO
-   (void)data;
-   return false;
+   int mib[6] = {CTL_NET,
+      PF_ROUTE, /* routing messages */
+      0, /* protocal number, currently always 0 */
+      0, /* select all address families */
+      NET_RT_IFLIST2, /* interface list with addresses */
+      0};
+   size_t len;
+
+   /* Determine len */
+   if (sysctl(mib, ARRAYSIZE(mib), NULL, &len, NULL, 0) < 0)
+      return false;
+
+   char *buf = xMalloc(len);
+
+   if (sysctl(mib, ARRAYSIZE(mib), buf, &len, NULL, 0) < 0) {
+      free(buf);
+      return false;
+   }
+
+   unsigned long long int bytesReceived_sum = 0, packetsReceived_sum = 0, bytesTransmitted_sum = 0, packetsTransmitted_sum = 0;
+
+   for (char *next = buf; next < buf + len;) {
+      IGNORE_WCASTALIGN_BEGIN
+      struct if_msghdr *ifm = (struct if_msghdr*) next;
+      IGNORE_WCASTALIGN_END
+
+      next += ifm->ifm_msglen;
```

Comment:
You can avoid the cast alignment warning by first casting to `void*` in an intermediate step.